resdk/query.py

```diff
@@ -239,6 +239,10 @@ def _fetch(self):
             raise NotImplementedError(
                 'Unsupported query_method: {}'.format(self.resource.query_method))
 
+        # Extract data from paginated response
+        if len(items) == 4 and set(items) == set(['next', 'previous', 'count', 'results']):
```

Comment:
To really make sure that response is paginated. If only check for `results` key is performed, strange errors can occurs if response contains `results` key by accident. But you are right, I have to add a check that `items` is a dict :)